client/my-sites/checkout/upsell-nudge/professional-email-upsell/index.tsx

```diff
@@ -17,30 +17,43 @@ import {
 	transformMailboxForCart,
 	validateMailboxes,
 } from 'calypso/lib/titan/new-mailbox';
+import type { MinimalRequestCartProduct } from '@automattic/shopping-cart';
+import type { TranslateResult } from 'i18n-calypso';
+import type { ChangeEvent } from 'react';
 
 import './style.scss';
 
-const noopWithCallback = ( ignored, callback = () => {} ) => {
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noopWithCallback = ( cartItem: MinimalRequestCartProduct, callback = () => {} ) => {
 	callback();
 };
 
-const ProfessionalEmailFeature = ( { children } ) => {
+const ProfessionalEmailFeature = ( { children }: { children: TranslateResult } ) => {
```

Comment:
Not related to your changes but we could rename `children` to `feature` or something a bit more specific.

Reply:
We're using `children` because the values are supplied as children of the component. While I agree that this should be updated, I don't want to tackle that in this PR -- I'll follow up once this PR lands.

Reply:
> We're using children because the values are supplied as children of the component.

Oh somehow I've missed that, maybe I didn't look up the right string when I searched. So `children` makes totally sense in that case.